Firestore/core/src/firebase/firestore/api/firestore.cc

```diff
@@ -146,6 +140,17 @@ void Firestore::Terminate(util::StatusCallback callback) {
   client_->TerminateAsync(std::move(callback));
 }
 
+void Firestore::Dispose() {
+  std::lock_guard<std::mutex> lock{mutex_};
+
+  // If the client hasn't been configured yet we don't need to create it just
+  // to dispose of it.
+  if (!client_) return;
+
+  client_->Dispose();
+  client_.reset();
```

Comment:
Question: what do you think about setting `extension_` to null here? It would have likely made the crash I've been seeing occur earlier and with a slightly more relevant stack trace. On the other hand, perhaps this is too specific to a particular problem.